src/main/java/seedu/address/model/TaskList.java

```diff
@@ -26,4 +26,24 @@ public int size() {
     public void deleteTask(Integer taskNumber) {
         taskList.remove(taskNumber - 1); // to convert to zero-based indexing
     }
+
+    /**
+     * Returns an ArrayList of matching tasks based on keyword.
+     *
+     * @param input the keyword input by user
+     * @return the list of tasks that contains the keyword
+     */
+    public ArrayList<Task> findTask(String input) {
+        ArrayList<Task> matchingTasks = new ArrayList<>();
+        for (int i = 0; i < this.taskList.size(); i++) {
+            Task curr = this.taskList.get(i);
+            if (curr.toString().contains(input)) {
```

Comment:
Perhaps there should be a choice for the user to choose if the task list should be searched based on the full input or can be part of the word as well. For instance, findt test and findt tes should give different results based on the users requirements on the search

Reply:
Also to add on, i believe `contains()` is case sensitive, might want to bypass that with regex

Reply:
Fixed with update. Choice for user will be updated.